transport.go

```diff
@@ -0,0 +1,63 @@
+package goftp
+
+import (
+	"crypto/tls"
+	"io"
+	"net/http"
+	"strings"
+)
+
+// Transport implements the http.RoundTripper interface.
+// Typical usage would be to register a Transport to handle
+// ftp:// and/or ftps:// URLs with http.Transport.RegisterProtocol.
+type Transport struct {
+	Config
+}
+
+// RoundTrip implements the http.RoundTripper interface to allow an http.Client
+// to handle ftp:// or ftps:// URLs.
+func (t *Transport) RoundTrip(req *http.Request) (*http.Response, error) {
+	config := t.Config
+	switch req.URL.Scheme {
+	default:
+		return nil, http.ErrSkipAltProtocol
+	case "ftp":
+	case "ftps":
+		if config.TLSConfig == nil {
+			config.TLSConfig = &tls.Config{}
+		}
+	}
+
+	if req.URL.User != nil {
+		if user := req.URL.User.Username(); user != "" {
```

Comment:
Is empty string user not a valid user name?

Reply:
You’re right, an empty string is a valid username. We have a choice: if `req.URL.User` is non-nil, then use the username and password from it, regardless of whether either is an empty string.

Reply:
https://golang.org/pkg/net/url/#Userinfo.Password has the `bool` 2nd return value.

Reply:
Done (62a7219 and 4c43d0e)